zbus/src/lib.rs

```diff
@@ -41,8 +41,20 @@ pub use dbus_error::*;
 mod error;
 pub use error::*;
 
-mod address;
-pub use address::*;
+pub mod address;
+
+#[cfg(any(
+    all(feature = "vsock", not(feature = "tokio")),
+    feature = "tokio-vsock"
+))]
+#[deprecated(note = "Use `address::VsockAddress` instead")]
+pub type VsockAddress = address::VsockAddress;
+#[deprecated(note = "Use `address::TcpAddress` instead")]
+pub type TcpAddress = address::TcpAddress;
```

Comment:
I think it's up to us.. not really sure that's necessary. For example, std::net has tcp stuff: https://doc.rust-lang.org/std/net/index.html

Reply:
True. Not sure, why i remembered wrong. :shrug: Since we're doing this, I think we might as well go all in and take full advantage of the module system.

Reply:
but I leave this to you. Please resolve this conversation if you think more strongly about it than me. :)

Reply:
also are you sure you want to replace something like Foo with foo::Foo ? (think Address, Message etc). I think the top user facing types could remain exported in the top namespace. Or we could use a prelude::* to help users...

Reply:
Hmm.. good point. We could move `Foo` under `foo` but still provide `Foo` on the top-level (w/o deprecation) but not for `foo::Bar`.

> Or we could use a prelude::* to help users...

I always failed to appreciate that pattern. Sure, it makes things easier for users but then it sacrifices readability too much IMO.

Reply:
yes, and you never now what is included or not in the prelude... it's just same a having it exported in the top-namespace imho in the end